src/include/catalog/catalog_accessor.h

```diff
@@ -269,6 +269,66 @@ class CatalogAccessor {
    */
   common::ManagedPointer<storage::BlockStore> GetBlockStore() const;
 
+  /**
+   * Adds a language to the catalog (with default parameters for now) if
+   * it doesn't exist in pg_language already
+   * @param lanname name of language
+   * @return oid of added language if it didn't exist before or INVALID_LANGUAGE_OID if else
+   */
+  language_oid_t CreateLanguage(const std::string &lanname);
```

Comment:
This seems like an insufficient prototype.  I would lean toward asking for everything we *will* need and dropping them unimplemented things inside the catalog as this is apart of the API and changing the signature will require refactoring more and more places as this gets leveraged.

Reply:
I did this as I don't think we intend on using this beyond just as a reference table for pg_proc anytime soon.

Reply:
If that’s the case I’d leave it out of catalog accessor and leave a todo in database catalog.  That should (hopefully) force someone to think about the end API before using it (because they’ll have to implement it).